src/components/progress-bar/EventProgressBar.tsx

```diff
@@ -0,0 +1,86 @@
+import { SliderProps } from '@mui/material/Slider/Slider';
+import { FC, useState } from 'react';
+
+import { EventRailsProvider } from '../../context/EventRailsProvider';
+import { useMediaListener } from '../../hooks';
+import { Highlight } from '../../types';
+import { EmitterListeners } from '../../types/emitters';
+import { blend, BlendColors, toTwoDigits } from '../../utils';
+import { PROGRESS_BAR_DIVIDER } from '../../utils/constants';
+
+import { EventRail } from './components/EventRail';
+import { ProgressBarStyled } from './components/ProgressBarStyled';
+
+interface EventProgressBarProps extends SliderProps {
+	mediaListener?: EmitterListeners;
+	setCurrentTime?: (relativeSeconds: number) => void;
+	highlights?: Highlight[];
+	getHighlightColorBlended?: BlendColors;
+}
+
+/** A MUI Slider configured for displaying currentTime/duration values from `MediaStore`
+ * uses `EmitterListeners` for displaying data
+ * @category React Component
+ * @category UI Controls
+ */
+export const EventProgressBar: FC<EventProgressBarProps> = ({
```

Comment:
Can we find a better name for the component? We have now `ProgressBar` and `EventProgressBar`, both with very similar comment to describe it. I would find it really hard to decide when to use what, without reading the details of the code.

Reply:
@ronnyroeller, maybe `EventBasedProgressBar` ?